include/net/lwm2m.h

```diff
@@ -137,6 +137,7 @@ struct lwm2m_ctx {
 	 */
 	char *desthostname;
 	uint16_t desthostnamelen;
+	bool hostname_verify;
```

Comment:
I think this is changing the default behavior now.

By default, samples seems to have this struct usually initialized to zero, which means that this bool would be false. Previously we set the hostname, so the default should be true.

It is unfortunate, if some MbedTLS config conflicts this and does not allow hostname checking, but that does not work the same with offloaded IP/TLS stacks. So we are changing behaviour of external TLS stack users.

Another thing: Why would you set the hostname, but set the flag to not verify it?
If application does not want hostname to be checked, it can pass a NULL pointer (leave it unititialized, if all are zero by default).

So I'm not in favour of this flag. I don't understand the need to name AND the flag, when same is achieved just by using the name pointer.

Reply:
That's the point we want to change the behaviour and disable setting of this option by default. `mbedtls_ssl_set_hostname()`, can only be used with X509 cetritificate support enabled in mbedTLS, and that's often not the case for LwM2M, where PSK is used (for example Zephyr sample does not enable cert support by default). This has been overlooked in the original PR adding hostname verification, and that's what we're trying to fix here.

The application has no control over desthostname/desthostnamelen as it's set by the engine while server URL is being parsed. There a flag is added informing the stack whether the stack should process the hostname or not. By the way, regardless of this PR, I think we need to do some cleanup in the LwM2M ctx struct to differentiate between internal and public fields, as we do in MQTT.

Reply:
OK, thank you for clarifying. I think I overlooked that detail

This is puzzling me.. I'm using offloaded TLS stack and PSK, still the server name extension is there
![image](https://user-images.githubusercontent.com/3104794/161041105-f252a14e-1f2e-44c6-a464-c5539af012df.png)

So maybe I'll just approve this. I'm not sure how we end  up documenting it. MbedTLS should really improve on that PSK use-case and allow checking for server name extensions even without X509 support.